### DISCUSSION TITLE
`ListViem` is lagging while the number of items scales up

### DISCUSSION THREAD

**Discussion by gusye1234** [Textualize/textual#2784] [Q&A] [answered]
Hi, I'm trying to implement a TXT reader based on `textual`. 
But I can experience significant display lagging while I have a `ListViem` to mount (but the size is only 955).

Is it normal? Or by design, `ListViem` is not supposed to be used in this case

**Comment by davep** [comment, marked as answer]
`ListItem` is really designed for lists on the smaller side, each item in the list is a widget in its own right (and in turn can contain multiple widgets). Adding lots of widgets will slow things down a wee bit.

Depending on what it is you're wanting to do, you'll probably find that [`OptionList`](https://textual.textualize.io/widgets/option_list/) is better suited to your needs here; it's just a single widget and can contain lots of content without any noticeable impact.
